src/main/java/org/folio/linked/data/mapper/dto/monograph/TopResourceMapperUnit.java

```diff
@@ -24,28 +21,16 @@ public Set<Class<?>> supportedParents() {
     return SUPPORTED_PARENTS;
   }
 
-  protected List<String> getPossibleLabels(List<TitleTitleInner> titles) {
+  protected List<String> getPrimaryMainTitles(List<TitleField> titles) {
     if (isNull(titles)) {
       return new ArrayList<>();
     }
     return titles.stream()
-      .sorted(Comparator.comparing(o -> o.getClass().getSimpleName()))
-      .map(TopResourceMapperUnit::getMainTitle).toList();
-  }
-
-  private static String getMainTitle(TitleTitleInner t) {
-    if (t instanceof BasicTitleField basicTitleField) {
-      var basicTitle = basicTitleField.getBasicTitle();
-      return getFirstValue(basicTitle::getMainTitle);
-    }
-    if (t instanceof ParallelTitleField parallelTitleField) {
-      var parallelTitle = parallelTitleField.getParallelTitle();
-      return getFirstValue(parallelTitle::getMainTitle);
-    }
-    if (t instanceof VariantTitleField variantTitleField) {
-      var variantTitle = variantTitleField.getVariantTitle();
-      return getFirstValue(variantTitle::getMainTitle);
-    }
-    return "";
+      .filter(PrimaryTitleField.class::isInstance)
+      .map(PrimaryTitleField.class::cast)
+      .map(PrimaryTitleField::getPrimaryTitle)
+      .filter(pt -> isNotEmpty(pt.getMainTitle()))
```

Comment:
minor
```
      .map(PrimaryTitleField::getPrimaryTitle)
      .map(pt -> pt.getMainTitle())
      .filter(CollectionUtils::isNotEmpty)
      .flatmap(Colletions::stream)
      .toList()
    ```
    the `filter` can be removed at all if the main title is always "non null"

Reply:
correct, fixed, thanks